cmd/kubeadm/app/apis/kubeadm/v1beta4/types.go

```diff
@@ -597,3 +597,134 @@ type Timeouts struct {
 	// +optional
 	Discovery *metav1.Duration `json:"discovery,omitempty"`
 }
+
```

Comment:
please apply all my requested changes that i did above in comments for kubeadm/types.go.